src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (!Animal.alive.includes(this)) {
+      Animal.alive.push(this);
+    }
+  }
+
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
```

Comment:
This constructor can be simplified. As 'hidden' property is specific to Herbivore and has an initial value, we don't need to pass it as a parameter.